clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -621,7 +622,8 @@ private void doCommitOffsetsAsync(final Map<TopicPartition, OffsetAndMetadata> o
             public void onSuccess(Void value) {
                 if (interceptors != null)
                     interceptors.onCommit(offsets);
-
+                offsets.forEach((tp, offsetMeta) ->
+                        offsetMeta.leaderEpoch().ifPresent(epoch -> metadata.updateLastSeenEpochIfNewer(tp, epoch)));
```

Comment:
I am not sure it makes sense to make updating the epoch conditional on the success of the offset commit. The coordinator does not do any validation of the value. I would expect that it would be handled similarly as the `seek` API.